api/datastore/bolt/bolt.go

```diff
@@ -239,15 +239,11 @@ func (ds *BoltDatastore) GetApp(ctx context.Context, name string) (*models.App,
 }
 
 func (ds *BoltDatastore) getRouteBucketForApp(tx *bolt.Tx, appName string) (*bolt.Bucket, error) {
-	var err error
 	// todo: should this be reversed?  Make a bucket for each app that contains sub buckets for routes, etc
 	bp := tx.Bucket(ds.routesBucket)
 	b := bp.Bucket([]byte(appName))
 	if b == nil {
-		b, err = bp.CreateBucket([]byte(appName))
```

Comment:
Can you verify we're not expecting this behavior anywhere else?  I think you're right that we shouldn't create an app when getting a route, but want to make sure anything using this method is ready for the change.